### PR TITLE
chore: update style dictionary

### PR DESCRIPTION
**Pull request by melaniebmn**
# Summary | Résumé

Updating Style Dictionary from version 3.9.2 to version 4.1.4. Style Dictionary introduced a lot of breaking changes with this major version release which required some updates to our config file.

To prepare for the rework of our typography tokens, which will happen soon, I took the opportunity to split them into separate files for easier maintenance and also cleaned up the code for the font sizes and line height calculations.

With this update, I was also able to remove the weird .css and .scss files that were being generated which we didn't use anywhere. In addition, we do not need to have the custom file header text anymore which prevented the timestamp from being generated for each build file. Style Dictionary finally doesn't include a timestamp in the file header output by default anymore, it is now an opt-in setting.

Apologies for the giant PR and amount of changed files - the removal of our custom file header meant that we are now going back to the default Style Dictionary header which included some extra text.

### CODE REVIEW

File: tokens/components/icon/tokens.js [hunk 1]
```diff
@@ -1,192 +1,67 @@
...
-    h2: {
-      value: '{fontSizes.h2.value}',
-      type: 'fontSizes',
-    },
-    h1: {
-      value: '{fontSizes.h1.value}',
-      type: 'fontSizes',
-    },
-  },
+  size: createPropertiesObject(fontSizeKeys, 'fontSizes'),
```
Comment (daine) [inline tokens/components/icon/tokens.js:27]: chef's kiss - i love deleting code

File: tokens/global/typography/lineHeights.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import typography from './core.js';
+import sizes from './fontSizes.js';
+
+const base = typography.base;
+const fontSizes = sizes.fontSizes;
+const headingSizes = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+const octave = 6;
```
Comment (daine) [inline tokens/global/typography/lineHeights.js:7]: Just curious here - who is the person who defined the octave (the person who did the manual work in figma), and could we have that referenced here if possible for historical purposes?

File: tokens/global/typography/lineHeights.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Calculates line height as a percentage based on font size.
+ *
+ * This function:
+ * - Takes font size as input and scales it relative to base font size.
+ * - Determines number of baseline octaves required to fit font size.
+ * - Adds a small additional octave to ensure adequate spacing.
+*/
+const calculateLineHeight = (fontSize) => {
+  const fontSizeValue = parseFloat(fontSize) / base.fontSize.value;
```
Comment (daine) [inline tokens/global/typography/lineHeights.js:18]: This function, if the `fontSize` is set to `base.fontSize.value` returns the same value as
```
base.lineHeight.value * 100
```

but we can simply skip all the other computation and just return the value if we're computing the line height for the base font size like this:

```suggestion
  const calculateLineHeight = (fontSize) => {
  if( fontSize === base.fontSize.value ) {
    return base.lineHeight.value * 100
  }
  const fontSizeValue = parseFloat(fontSize) / base.fontSize.value;
```

File: tokens/global/typography/lineHeights.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+});
+
+// Initialize line heights object
+const lineHeights = {
+  caption: createLineHeightObject('caption', fontSizes.caption.value),
+  captionMobile: createLineHeightObject('captionMobile', fontSizes.captionMobile.value),
+  text: {
+    value: `${base.lineHeight.value * 100}%`,
+    type: 'lineHeights',
+  },
```
Comment (daine) [inline tokens/global/typography/lineHeights.js:43]: Could we also change this to move the computation in the function? See comment above!

**Review by daine — CHANGES_REQUESTED** [review]
Just more thing I noticed, but it looks great!

File: tokens/global/typography/lineHeights.js (hunk 2 above)
Comment (melaniebmn) [reply tokens/global/typography/lineHeights.js:7]: It was Phil when he was briefly on our team. I will rewrite this part when I implement the new typography and spacing so we can replace the octave.

File: tokens/components/icon/tokens.js (hunk 1 above)
Comment (melaniebmn) [reply tokens/components/icon/tokens.js:27]: Me too :)

File: tokens/global/typography/lineHeights.js (hunk 3 above)
Comment (melaniebmn) [reply tokens/global/typography/lineHeights.js:18]: I will rewrite the line height calculation to implement the new typography. Would you be okay with leaving it as is for now?

Comment (daine) [reply tokens/global/typography/lineHeights.js:18]: Yes sounds good to me!

**Review by daine — APPROVED** [review]
LGTM!

**Pull request merged**